### ISSUE TITLE
Question about the app + MacOS support

### ISSUE THREAD

**Issue by mjehrhart** [YassinEldeeb/deadliner#6]
Does your app run as background wallpaper for the desktop?

Thanks.

**Comment by YassinEldeeb** [comment]
Yes exactly, It runs in the system tray as a Cron scheduler depending on what is the least time unit you choose to show as a time format, It loads the image you choose whether it's a solid color, cached downloaded image from a URL or an image from disk and draws the time left with your chosen font at the center of the image then set it as a wallpaper for your desktop, this operation is supposed to be very efficient than casual ideas like widgets, standalone apps, ..etc. So that's the big deal about it and btw there are still a lot of optimizations that should be done for improving CPU & Memory usage even more(The goal is to give fancy features like setting deadlines in a gentle way for super low-end hardware users cause two years ago I struggled with using these fancy apps on my toaster laptop).

Btw, I should enable discussions on this repo for good discussions like this.

Thank you for your interest to know more about the app, If you want to know any other details I'm here to help in the discussions tab.

**Comment by YassinEldeeb** [comment]
How did you find the project btw? I'm curious to know.

**Comment by mjehrhart** [comment]
Thanks for sharing the details. I'm on a Mac and so cannot test.  But it think its a great idea and you did a good job showing it off.  Do you think this could be done on a Mac?

I found your repo from emilk/egui discussions.

**Comment by YassinEldeeb** [comment]
## > I'm on a Mac and so cannot test, Do you think this could be done on a Mac?

I'm pretty sure that I've ensured that everything is cross-platform compatible, but unfortunately, I don't have access to a Mac, so I've only tested it on Windows/Linux which works pretty well, If you can share with me what problems you've faced when trying it, I think I would be able to support Mac from your feedback.

You can download it for Mac [here](https://github.com/YassinEldeeb/Deadliner/releases/tag/v0.1.5-beta).

### How to run it?
I think binaries on Mac don't have an extension so you would have to spin up a terminal and execute it from there with the needed permissions, you should be able to see the GUI running, If not please let me know.

## > you did a good job showing it off.

Thank you ❤, yes I've made everything that I can do to show it off; [video](https://www.youtube.com/watch?v=ySn-Ykdq5BU&feature=youtu.be), [website](https://deadliner.vercel.app/), [discussions](https://github.com/emilk/egui/issues/996)

## > I found your repo from emilk/egui discussions.

Oh yeah, I've also seen [your project](https://github.com/mjehrhart/minty) too there, It's very cool 👏.

**Comment by YassinEldeeb** [comment]
Also if you were a bit curious about why the app consists of two separate binaries 🤔
![image](https://user-images.githubusercontent.com/72823042/164945822-4590eba0-7162-4c5a-812e-f31748cea561.png)

It's because Egui doesn't support killing the window without exiting the whole process as discussed [here](https://github.com/emilk/egui/issues/1223) and that's a problem cause the app needs to run in the system tray without the GUI and It needs the functionality to close/open the GUI without exiting the scheduler process, so that's why they're two separate binaries and they communicate through an HTTP server running on the scheduler binary to ping a binary to shutdown or check the status of a running process.
![Screenshot 2022-04-23 230247](https://user-images.githubusercontent.com/72823042/164945896-06cb6692-a586-4e4a-ad1d-ca7eb2bca83a.png)
